### PR TITLE
os2web_key

### PR DESCRIPTION
**Pull request by rimi-itk**
Adds `os2web_key` module. See <https://github.com/OS2web/os2web_key/blob/os2web_key/README.md> for details.

### CODE REVIEW

File: src/Exception/RuntimeException.php [hunk 1]
```diff
@@ -0,0 +1,10 @@
+<?php
+
+namespace Drupal\os2web_key\Exception;
+
+/**
+ * Runtime exception.
+ */
+class RuntimeException extends \RuntimeException {
```
Comment (cableman) [inline src/Exception/RuntimeException.php:8]: Would it be better to rename this exception to something like `ModuleRuntimeExeception` or `SSLRuntimeException`?

File: src/Plugin/KeyType/CertificateKeyType.php [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+    ];
+
+    return $form;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function validateConfigurationForm(array &$form, FormStateInterface $form_state): void {
+  }
```
Comment (cableman) [inline src/Plugin/KeyType/CertificateKeyType.php:113]: Empty validation function, would it be possible to validate the `Passphrase` for the certificate in some way if one is given?

**Review by cableman — CHANGES_REQUESTED** [review]

File: src/Plugin/KeyType/CertificateKeyType.php (hunk 2 above)
Comment (rimi-itk) [reply src/Plugin/KeyType/CertificateKeyType.php:113]: Actual validation is performed in `validateKeyValue`. I've added a comment telling this in https://github.com/OS2web/os2web_key/pull/1/commits/0eb16ad83c61f66d80da0e795314dbaf3672216b.

File: src/Exception/RuntimeException.php (hunk 1 above)
Comment (rimi-itk) [reply src/Exception/RuntimeException.php:8]: That would not add any value. The `RuntimeException` is namespaced and can be aliased (`use … as …`) if need be.

**Review by cableman — APPROVED** [review]

**Pull request merged**